clusterloader2/pkg/measurement/common/slos/api_responsiveness_prometheus.go

```diff
@@ -204,13 +224,43 @@ func (a *apiResponsivenessGatherer) gatherAPICalls(executor QueryExecutor, start
 	return newFromSamples(latencySamples, countSamples, countSlowSamples)
 }
 
-func (a *apiResponsivenessGatherer) validateAPICalls(identifier string, allowedSlowCalls int, metrics *apiCallMetrics) []error {
+func getCustomThresholds(config *measurement.Config, metrics *apiCallMetrics) (customThresholds, error) {
+	thresholdsString, err := util.GetStringOrDefault(config.Params, "customThresholds", "")
+	if err != nil {
+		return nil, err
+	}
+	var thresholds []customThresholdEntry
+	if err := yaml.Unmarshal([]byte(thresholdsString), &thresholds); err != nil {
+		return nil, err
+	}
+
+	ct := customThresholds{}
+	for _, entry := range thresholds {
+		if entry.Threshold == 0 {
+			return nil, fmt.Errorf("custom threshold must be set to a positive time duration")
+		}
+		key := entry.getKey()
+		if _, ok := metrics.metrics[key]; !ok {
+			return nil, fmt.Errorf("unrecognized custom threshold API call key: %v", key)
```

Comment:
Let's log warning instead of failing the test here

Reply:
Done.